plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorListener.kt

```diff
@@ -29,7 +31,11 @@ class CodeWhispererEditorListener : EditorFactoryListener {
                 // the most accurate code percentage data.
                 override fun documentChanged(event: DocumentEvent) {
                     if (!isCodeWhispererEnabled(project)) return
-                    CodeWhispererInvocationStatus.getInstance().documentChanged()
+                    if (CodeWhispererFeatureConfigService.getInstance().getNewAutoTriggerUX()) {
+                        CodeWhispererInvocationStatusNew.getInstance().documentChanged()
+                    } else {
+                        CodeWhispererInvocationStatus.getInstance().documentChanged()
```

Comment:
for all these singleton service, I think we can do 
```
fun getInstance(): CodeWhispererService = if (newUX) CodeWhispererServiceNew.getInstance() else service()
```

Reply:
CodeWhispererServiceNew.getInstance() is of type `CodeWhispererServiceNew.getInstance()`, so the `getInstance()` will have a type of Any then